README.md

```diff
@@ -16,15 +17,138 @@ TerseTS can be compiled and cross-compiled from source:
    - macOS: `zig build -Dtarget=aarch64-macos -Doptimize=ReleaseFast`
    - Microsoft Windows: `zig build -Dtarget=x86_64-windows -Doptimize=ReleaseFast`
 
-# Usage
-TODO
+## Usage
+TerseTS provides a C-API that is designed to be simple to wrap. Currently, TerseTS includes APIs for the following programming languages which can be used without installation of any dependencies:
+<a id="zig-usage-example"></a>
+<details>
+<summary><strong>Zig Usage Example</strong></summary>
 
-# Bindings
-TerseTS provides a C-API that is designed to be simple to wrap. Currently, TerseTS includes bindings for the following programming languages which can be used without installation of any dependencies:
-- [Zig](src/tersets.zig)
-- [C](bindings/c/tersets.h)
-- [C++](bindings/c/tersets.h)
-- [Python](bindings/python/tersets) using [ctypes](https://docs.python.org/3/library/ctypes.html)
+```c
+const std = @import("std");
+const tersets = @import("path/to/tersets.zig");
+
+pub fn main() void {
+   var uncompressed_values = [_]f64{1.0, 2.0, 3.0, 4.0, 5.0};
+   std.debug.print("Uncompressed data length: {any}\n", .{uncompressed_values.len});
+   
+   // The supported compression methods are specified in tersets.zig.
+   const config = tersets.Configuration{
+        .method = .SwingFilter,
+        .error_bound = 0.1,
+   };
+    
+   var compressed_values = try tersets.compress(data[0..], config);
+   defer std.heap.page_allocator.free(compressed);
+
+   std.debug.print("Compression successful. Compressed data length: {any}\n", .{compressed_values.items.len});
+    
+   var decompressed_values = try tersets.decompress(compressed, config);
+   defer std.heap.page_allocator.free(decompressed);
+
+   std.debug.print("Decompression successful. Decompressed data length {any}\n", .{decompressed_values.items.len});
+}
+```
+
+TerseTS provides `./src/tersets.zig` as the single access point and two main functions `compress` and `decompress`. 
+
+For compression, you can select the compression method through the `Configuration` structure with two parameters: the compression method, e.g., `.method=.SwingFilter`, and the error bound, e.g., `.error_bound = 0.1`. The supported compression methods are specified in `src/tersets.zig`. 
```

Comment:
Maybe make use of spaces around `=` consistent through the README.md. From a quick skim, it seems they are mostly included.

Reply:
I see, good catch. Fixed.